src/addrman.h

```diff
@@ -370,14 +382,34 @@ friend class CAddrManTest;
         s << asmap_version;
     }
 
-    template<typename Stream>
-    void Unserialize(Stream& s)
+    template <typename Stream>
+    void Unserialize(Stream& s_)
     {
         LOCK(cs);
 
         Clear();
-        unsigned char nVersion;
-        s >> nVersion;
+
+        Format format;
+        s_ >> Using<CustomUintFormatter<1>>(format);
+
+        static constexpr Format maximum_supported_format = Format::V3_BIP155;
+        if (format > maximum_supported_format) {
+            throw std::ios_base::failure(strprintf(
+                "Unsupported format of addrman database: %u. Maximum supported is %u. "
+                "Continuing operation without using the saved list of peers.",
```

Comment:
6fce3ce maybe
```suggestion
                "Continuing operation with a new peers.dat in the maximum supported format",
```
(and without the full stop), as currently it would print

`ERROR: DeserializeDB: Deserialize or I/O error - Unsupported format of addrman database: 4. Maximum supported is 3. Continuing operation without using the saved list of peers.: iostream error`

Reply:
"Continuing operation by creating a new peers.dat" is, strictly speaking, not true because the new `peers.dat` will not be created at the time this message is printed. It will be created later during shutdown or during one of the periodic dumps to disk.

I left it as is.